services/movie_session.py

```diff
@@ -0,0 +1,57 @@
+from datetime import datetime
+
+from django.db.models import QuerySet
+
+from db.models import MovieSession
+
+
+def create_movie_session(
+        movie_show_time: str,
+        movie_id: int,
+        cinema_hall_id: int
+) -> MovieSession:
+    return MovieSession.objects.create(
+        show_time=movie_show_time,
+        cinema_hall_id=cinema_hall_id,
+        movie_id=movie_id
+    )
+
+
+def get_movies_sessions(
+        session_date: str | None = None
+) -> QuerySet[MovieSession]:
+    if session_date:
+        if len(session_date) == 19:
+            date = datetime.strptime(session_date, "%Y-%m-%d %H:%M:%S")
+            return MovieSession.objects.filter(show_time=date)
+        else:
+            date = datetime.strptime(session_date, "%Y-%m-%d")
+            return MovieSession.objects.filter(show_time__date=date)
+
+    return MovieSession.objects.all()
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: str | None = None,
+        movie_id: int | None = None,
+        cinema_hall_id: int | None = None
+) -> QuerySet:
+    movie_session = MovieSession.objects.filter(id=session_id)
+
+    if show_time:
+        movie_session.update(show_time=show_time)
+    if movie_id:
+        movie_session.update(movie_id=movie_id)
+    if cinema_hall_id:
+        movie_session.update(cinema_hall_id=cinema_hall_id)
+
+    return movie_session
```

Comment:
```suggestion
    movie_session = get_movie_session_by_id(movie_session_id =session_id)
    if show_time:
        movie_session.show_time = show_time
    if movie_id:
        movie_session.movie_id = movie_id
    if cinema_hall_id:
        movie_session.cinema_hall_id = cinema_hall_id
    return movie_session.save()
```

this reduces the number of requests to the DB, because we do not need to make three update requests